### PR TITLE
cli: Don't fail clone if we have repo locally

### LINKED ISSUE #311: `rad-clone`: Don't create fork if we already have a fork
Currently we don't check, so it probably fails if we're cloning something for the second time.

### LINKED ISSUE #338: `rad clone` fails in a singe-node scenario with `failed no seeds found`
https://github.com/radicle-dev/heartwood/commit/e584bd6c1cbb5c8bb9b74a856f6fdd7c22392efe: If you push a project to your node and then try to clone it from same profile/node, you'll encounter:

```bash
== Clone failed no seeds found for rad:z5mjpEsLp9yLYyD5gez85eamUEmk
```

this problem doesn't exist in a multi-node setup.

### PR DESCRIPTION
**Pull request by cloudhead**
Fixes #338
Fixes #311

### CODE REVIEW

**Review by FintanH — APPROVED** [review]
Makes sense :+1:

File: radicle-cli/src/commands/clone.rs [hunk 1]
```diff
@@ -175,31 +175,36 @@ pub fn clone<G: Signer>(
...
+            match node.fetch(id, *seed)? {
+                FetchResult::Success { .. } => {
+                    spinner.finish();
+                }
+                FetchResult::Failed { reason } => {
+                    spinner.error(reason);
+                }
             }
         }
+        // TODO: Warn if no seeds were found, and we might be checking out stale data.
```
Comment (xphoniex) [inline radicle-cli/src/commands/clone.rs:198]: might also be helpful to assign someone to TODOs in comments as well, e.g. `TODO(dave)`

**Review by xphoniex — APPROVED** [review]
Looks good, tested locally and it works.

File: radicle-cli/src/commands/clone.rs (hunk 1 above)
Comment (cloudhead) [reply radicle-cli/src/commands/clone.rs:198]: Yeah we can start doing that.

Comment (xphoniex) [reply radicle-cli/src/commands/clone.rs:198]: works well with `grep`, and even ci/bots in future.

**Pull request merged**